torchbenchmark/models/pyhpc_equation_of_state/__init__.py

```diff
@@ -0,0 +1,53 @@
+import torch
+from . import eos_pytorch
+from ...util.model import BenchmarkModel
+
+def _generate_inputs(size):
+    import math
+    import numpy as np
+    np.random.seed(17)
+
+    shape = (
+        math.ceil(2 * size ** (1/3)),
+        math.ceil(2 * size ** (1/3)),
+        math.ceil(0.25 * size ** (1/3)),
+    )
+
+    s = torch.from_numpy(np.random.uniform(1e-2, 10, size=shape))
+    t = torch.from_numpy(np.random.uniform(-12, 20, size=shape))
+    p = torch.from_numpy(np.random.uniform(0, 1000, size=(1, 1, shape[-1])))
+    return s, t, p
+
+class EquationOfState(torch.nn.Module):
+    def __init__(self):
+        super(EquationOfState, self).__init__()
+
+    def forward(self, s, t, p):
+        return eos_pytorch.gsw_dHdT(s, t, p)
+
+class Model(BenchmarkModel):
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+        self.model = EquationOfState()
```

Comment:
You need to manually move the model to the appropriate device (this has been the source of a bunch of issues recently).

Reply:
On that note, I'd also be interested in knowing what the CUDA results are :P

Reply:
Done, and done!